erpnext/projects/doctype/project/project.py

```diff
@@ -322,17 +322,42 @@ def update_sales_amount(self):
 		self.total_sales_amount = total_sales_amount and total_sales_amount[0][0] or 0
 
 	def update_billed_amount(self):
+		# nosemgrep
+		self.total_billed_amount = (
+			self.get_total_billed_amount_from_parent() + self.get_total_billed_amount_from_child()
```

Comment:
This is not equivalent at all. You're double counting invoices where `project` is set on parent and children both. 

It should be possible to do this in same query without letting go of `if`. Try adding extra conditions to speed up the query but not double count things.